app/services/scheduled_publishing_failed_service.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class ScheduledPublishingFailedService
+  def call(edition_id)
+    edition = nil
+
+    Edition.transaction do
+      edition = Edition.lock.find_current(id: edition_id)
+      update_status(edition)
+    end
+
+    notify_editors(edition)
+  end
+
+private
+
+  def update_status(edition)
+    raise "Expected edition to be scheduled" unless edition.scheduled?
```

Comment:
Could you help me understand which edge case are we trying to protect against here? I'm still worried about this kind of statement proliferating into every class, without any definite value.

Reply:
Understand your concern. I think this one is valid as there is nothing to ensure the state of this edition.

Consider:

- Final scheduling fails and we leave this transaction: https://github.com/alphagov/content-publisher/blob/handle-publishing-failing/app/jobs/scheduled_publishing_job.rb#L22
- Some period of time occurs between this and https://github.com/alphagov/content-publisher/blob/handle-publishing-failing/app/jobs/scheduled_publishing_job.rb#L6
- In that time any changes to the state of edition would make this failed to publish make the data model inconsistent

It is super unlikely as the period of time is super short, but all it takes is a request to occur during the attempt of scheduling and that request would have the next lock on the edition.

Reply:
OK. It's good to see a potential scenario, and it helps me to think about this as a state machine transition - in that sense, we should be validating the origin (scheduled?) in order to transition to the destination (failed_to_publish).